echopype/consolidate/api.py

```diff
@@ -459,7 +463,11 @@ def add_splitbeam_angle(
     if "channel" not in source_Sv.variables:
         raise ValueError("The input source_Sv Dataset must have a channel dimension!")
 
-    # Select ds_beam channels from source_Sv
+    # Swap the dimension to frequency_nominal if use_frequency_nominal is True
+    if use_frequency_nominal:
+        source_Sv = swap_dims_channel_frequency(source_Sv)
```

Comment:
What I was thinking was to make the `add_splitbeam_angle` function agnostic of whether in the input `ds_Sv` dataset has a dimension of `channel` or `frequency_nominal`. So the function will work with the original dimension (`channel`) straight from `compute_Sv`, and will also work if the dimension has already been changed to `frequency_nominal` before the `add_splitbeam_angle` function is called. So not running the `swap_dims_channel_frequency` under the hood within this function, but made this functions accepts `ds_Sv` with either dimensions.

Does the above make sense to you?